src/effect.js

```diff
@@ -0,0 +1,56 @@
+define(function (require) {
+	'use strict';
+
+	var p5sound = require('master');
+
+	p5.Effect = function() {
+		
+		this.ac = p5sound.audiocontext;
+
+		this.input = this.ac.createGain();
+		this.output = this.ac.createGain();
+
+		this.dry = this.ac.createGain();
+		this.wet = this.ac.createGain();
+
+		this.input.connect(this.output);
+
+		this.connect();
+
+		p5sound.soundArray.push(this);
+	};
+
+
+	p5.Effect.prototype.connect = function (unit) {
+		var u = unit || p5.soundOut.input;
+		this.output.connect(u.input ? u.input : u);
+	};
+
+	p5.Effect.prototype.disconnect = function() {
+		this.output.disconnect();
+	};
+
+	p5.Effect.prototype.dispose = function() {
+		var index = p5sound.soundArray.indexOf(this);
+		p5sound.soundArray.splice(index, 1);
+
+		this.input.disconnect();
+		this.input = undefined;
+
+		this.output.disconnect();
+		this.output = undefined;
+	};
+
+	p5.Effect.prototype.dispose = function() {
+		var index = p5sound.soundArray.indexOf(this);
+		p5sound.soundArray.splice(index, 1);
+
+		this.input.disconnect();
+		this.input = undefined;
+
+		this.output.disconnect();
+		this.output = undefined;
+	};
+
+
```

Comment:
`return p5.Effect;` here so that Effect can be explicitly required by other modules